File: javascript/leetcode/find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    var sum = 0;
    var leftSum = 0
    for (var i = 0; i < nums.length; i++) {
        sum += nums[i]
    }
    for (var i = 0; i < nums.length; i++) {
        if (leftSum == sum - leftSum - nums[i]) return i;
        leftSum += nums[i]
    }
    return -1
};

File: ruby/leetcode/find-pivot-index.rb
# @param {Integer[]} nums
# @return {Integer}
def pivot_index(nums)
    sum = 0
    left_sum = 0
    nums.each do |n|
        sum += n
    end
    for i in 0...nums.size()
        n = nums[i]
        if left_sum == sum - left_sum - n
            return i
        end
        left_sum += n
    end
    return -1
end
